src/software/ai/hl/stp/tactic/crease_defender/crease_defender_fsm.cpp

```diff
@@ -91,18 +109,53 @@ void CreaseDefenderFSM::blockThreat(
     TbotsProto::BallCollisionType ball_collision_type =
         TbotsProto::BallCollisionType::ALLOW;
     if ((event.common.world_ptr->ball().position() - destination).length() <
-        (event.common.robot.position() - destination).length())
+        (robot_position - destination).length())
     {
         ball_collision_type = TbotsProto::BallCollisionType::AVOID;
     }
+    /**
+     * TODO: Fix Auto chip behaviour by adding conditional logic based on the current world
+     *  1) AutoChipOrKickMode::OFF when facing the net
+     *  2) AutoChipOrKickMode::AUTOCHIP <--> when near OUR net && enemy in front
+     *  3) AutoChipOrKickMode::OFF + DribbleFSM <--> when ball nearby && no nearby threats
+     */
+    AutoChipOrKick auto_chip_or_kick{};
+    auto goal_line_segment = Segment(event.common.world_ptr->field().friendlyGoal().posXPosYCorner(),
+                                     event.common.world_ptr->field().friendlyGoal().posXNegYCorner());
+    Ray robot_shoot_ray = Ray(robot_position, robot_orientation);
+    std::vector<Point> goal_intersections = intersection(robot_shoot_ray, goal_line_segment);
+    Stadium threat_zone = Stadium(robot_position,
+                                  Vector::createFromAngle(robot_orientation).normalize(1),
+                                  0.1);
+    double robot_to_net_m = distance(robot_position, event.common.world_ptr->field().friendlyGoal().centre());
+
+    // DEBUG: Visualizes threat stadium
+    LOG(VISUALIZE) << *createDebugShapes({
+         *createDebugShape(threat_zone, std::to_string(event.common.robot.id()), "threatzone")
+    });
+
+    if (goal_intersections.empty()
+        && CreaseDefenderFSM::isAnyEnemyInZone(event, threat_zone)
+        && robot_to_net_m <= event.common.world_ptr->field().totalYLength() / 2)
```

Comment:
why did you use YLength() here?

also this should be a constant with the word `THRESHOLD` in it

Reply:
This ensures that chips only occur when the ball is somewhat near our net. This cannot be a constant since this value must scale with field size and this is only determined in runtime